File: src/PointwiseFunctions/Hydro/EquationsOfState/Python/PolytropicFluid.cpp
// Distributed under the MIT License.
// See LICENSE.txt for details.

#include "PointwiseFunctions/Hydro/EquationsOfState/Python/PolytropicFluid.hpp"

#include <pybind11/pybind11.h>
#include <string>

#include "PointwiseFunctions/Hydro/EquationsOfState/PolytropicFluid.hpp"

namespace py = pybind11;

namespace EquationsOfState::py_bindings {

void bind_polytropic_fluid(py::module& m) {
  py::class_<PolytropicFluid<true>, EquationOfState<true, 1>>(
      m, "RelativisticPolytropicFluid")
      .def(py::init<double, double>(), py::arg("polytropic_constant"),
           py::arg("polytropic_exponent"));
}

}  // namespace EquationsOfState::py_bindings

File: tests/Unit/PointwiseFunctions/Hydro/EquationsOfState/PolytropicFluid.py
# Distributed under the MIT License.
# See LICENSE.txt for details.


def polytropic_pressure_from_density(
    rest_mass_density, polytropic_constant, polytropic_exponent
):
    return polytropic_constant * rest_mass_density**polytropic_exponent


def polytropic_rel_rest_mass_density_from_enthalpy(
    specific_enthalpy, polytropic_constant, polytropic_exponent
):
    return (
        (polytropic_exponent - 1.0)
        / (polytropic_constant * polytropic_exponent)
        * (specific_enthalpy - 1.0)
    ) ** (1.0 / (polytropic_exponent - 1.0))


def polytropic_newt_rest_mass_density_from_enthalpy(
    specific_enthalpy, polytropic_constant, polytropic_exponent
):
    return (
        (polytropic_exponent - 1.0)
        / (polytropic_constant * polytropic_exponent)
        * (specific_enthalpy)
    ) ** (1.0 / (polytropic_exponent - 1.0))


def polytropic_rel_specific_enthalpy_from_density(
    rest_mass_density, polytropic_constant, polytropic_exponent
):
    return 1.0 + polytropic_constant * polytropic_exponent / (
        polytropic_exponent - 1.0
    ) * rest_mass_density ** (polytropic_exponent - 1.0)


def polytropic_newt_specific_enthalpy_from_density(
    rest_mass_density, polytropic_constant, polytropic_exponent
):
    return (
        polytropic_constant
        * polytropic_exponent
        / (polytropic_exponent - 1.0)
        * rest_mass_density ** (polytropic_exponent - 1.0)
    )


def polytropic_specific_internal_energy_from_density(
    rest_mass_density, polytropic_constant, polytropic_exponent
):
    return (
        polytropic_constant
        * rest_mass_density ** (polytropic_exponent - 1.0)
        / (polytropic_exponent - 1.0)
    )


def polytropic_chi_from_density(
    rest_mass_density, polytropic_constant, polytropic_exponent
):
    return (
        polytropic_constant
        * polytropic_exponent
        * rest_mass_density ** (polytropic_exponent - 1.0)
    )


def polytropic_kappa_times_p_over_rho_squared_from_density(
    rest_mass_density, polytropic_constant, polytropic_exponent
):
    return 0.0
